react-cors-spa/src/AddProduct.jsx

```diff
@@ -0,0 +1,145 @@
+import * as React from 'react';
+
+import { Button, Skeleton } from "@mui/material";
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+import TextField from '@mui/material/TextField';
+import MenuItem from '@mui/material/MenuItem';
+import Select from '@mui/material/Select';
+import FormControl from '@mui/material/FormControl';
+import Box from '@mui/material/Box';
+
+import "./AddProductButton.css";
+import "./ProductDialogue.css";
+import "./AddProduct.css";
+
+import axios from 'axios';
+import { useQuery } from '@tanstack/react-query';
+
+import { useForm } from 'react-hook-form';
+
+function fetchCategories() {
```

Comment:
Let's start the pattern of putting these API methods in their own files. By the time we're done, we may want to call the same methods from multiple components. Usually these methods are grouped by table into "services". So you could put this one in src/services/CategoriesService.js.